src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -435,17 +489,52 @@ SegmentCleaner::init_segments_ret SegmentCleaner::init_segments() {
 	auto segment_id = it.first;
 	return scanner->read_segment_header(
 	  segment_id
-	).safe_then([&segment_set, segment_id, this](auto header) {
+	).safe_then([&segment_set, segment_id, this](auto header)
+	  -> ExtentReader::scan_extents_ertr::future<> {
 	  if (header.out_of_line) {
-	    logger().debug(
+	    logger().info(
 	      "ExtentReader::init_segments: out-of-line segment {}",
 	      segment_id);
-	    init_mark_segment_closed(
-	      segment_id,
-	      header.journal_segment_seq,
-	      true);
+	    //TODO: iterating through all extents of all ool segments to confirm their
+	    //	  last_modified time is very inefficient, should be improved
```

Comment:
This doesn't seem necessary.  Can we just use the time at which the segment was opened?  We can write that into the segment header on open.

Reply:
I think a segment may keep open for a relatively long time, so maybe we can add a segment tail in which we keep the time of the last modification time made to the segment?

Actually, when I was designing the paddr->laddr mapping mechinary, I thought maybe we can keep the modification time in the backref extents, what do you think?:-)

Reply:
A segment tail would work.  Putting the modification time in the backref extent seems semantically identical to putting it in the laddr map, either way it's a lot of overhead.